src/utils/utils.cpp

```diff
@@ -634,7 +634,62 @@ std::string mp::utils::emit_yaml(const YAML::Node& node)
 
 std::string mp::utils::emit_cloud_config(const YAML::Node& node)
 {
-    return fmt::format("#cloud-config\n{}\n", emit_yaml(node));
+    return fmt::format("#cloud-config\n{}", emit_yaml(node));
```

Comment:
This change is not particularly related to the purpose of this PR, it is just a small improvement that removes the 2nd redundant (emit_yaml already appended /n) new line.